geode-core/src/main/java/org/apache/geode/internal/cache/PartitionedRegionDataStore.java

```diff
@@ -1472,6 +1472,19 @@ public boolean removeBucket(int bucketId, boolean forceRemovePrimary) {
       }
 
       BucketAdvisor bucketAdvisor = bucketRegion.getBucketAdvisor();
+      InternalDistributedMember primary = bucketAdvisor.getPrimary();
+      InternalDistributedMember myId =
+          this.partitionedRegion.getDistributionManager().getDistributionManagerId();
+      if (primary == null || myId.equals(primary)) {
```

Comment:
This seems similar to the logic a few lines down where we say "if (!forceRemovePrimary && bucketAdvisor.isPrimary()) {..."

Unlike that line, your new logic doesn't honor the forceRemovePrimary flag. Should it? I don't actually see any cases where that is passed in as true, so maybe we should just remove that flag?